storm-core/src/jvm/org/apache/storm/daemon/supervisor/SupervisorData.java

```diff
@@ -0,0 +1,339 @@
+        }
+    }
+
+    public ConcurrentHashMap<String, List<ProfileRequest>> getStormIdToProfileActions() {
+        return stormIdToProfileActions;
+    }
+
+    public void setStormIdToProfileActions(Map<String, List<ProfileRequest>> stormIdToProfileActions) {
+        this.stormIdToProfileActions.clear();
+        this.stormIdToProfileActions.putAll(stormIdToProfileActions);
+    }
+
+    public Map getConf() {
+        return conf;
+    }
+
+    public void setConf(Map conf) {
+        this.conf = conf;
+    }
+
+    public IContext getSharedContext() {
+        return sharedContext;
+    }
+
+    public void setSharedContext(IContext sharedContext) {
+        this.sharedContext = sharedContext;
+    }
+
+    public boolean isActive() {
+        return active;
+    }
+
+    public void setActive(boolean active) {
+        this.active = active;
+    }
+
+    public ISupervisor getiSupervisor() {
+        return iSupervisor;
+    }
+
+    public void setiSupervisor(ISupervisor iSupervisor) {
+        this.iSupervisor = iSupervisor;
+    }
+
+    public Utils.UptimeComputer getUpTime() {
+        return upTime;
+    }
+
+    public void setUpTime(Utils.UptimeComputer upTime) {
+        this.upTime = upTime;
+    }
+
+    public String getStormVersion() {
+        return stormVersion;
+    }
+
+    public void setStormVersion(String stormVersion) {
+        this.stormVersion = stormVersion;
+    }
+
+    public ConcurrentHashMap<String, String> getWorkerThreadPidsAtom() {
+        return workerThreadPidsAtom;
+    }
+
+    public void setWorkerThreadPidsAtom(ConcurrentHashMap<String, String> workerThreadPidsAtom) {
+        this.workerThreadPidsAtom = workerThreadPidsAtom;
+    }
+
+    public IStormClusterState getStormClusterState() {
+        return stormClusterState;
+    }
+
+    public void setStormClusterState(IStormClusterState stormClusterState) {
+        this.stormClusterState = stormClusterState;
+    }
+
+    public LocalState getLocalState() {
+        return localState;
+    }
+
+    public void setLocalState(LocalState localState) {
+        this.localState = localState;
+    }
+
+    public String getSupervisorId() {
+        return supervisorId;
+    }
+
+    public void setSupervisorId(String supervisorId) {
+        this.supervisorId = supervisorId;
+    }
+
+    public String getAssignmentId() {
+        return assignmentId;
+    }
+
+    public void setAssignmentId(String assignmentId) {
+        this.assignmentId = assignmentId;
+    }
+
+    public String getHostName() {
+        return hostName;
+    }
+
+    public void setHostName(String hostName) {
+        this.hostName = hostName;
+    }
+
+    public ConcurrentHashMap<Long, LocalAssignment> getCurrAssignment() {
+        return currAssignment;
+    }
+
+    public void setCurrAssignment(Map<Long, LocalAssignment> currAssignment) {
+        this.currAssignment.clear();
+        this.currAssignment.putAll(currAssignment);
```

Comment:
The clear with the add should probably be atomic.  The race is very small, but I would rather see us have an AtomicReference<Map<Long, LocalAssignment>> instead of a ConcurrentHashMap.